packages/i18n/src/index.js

```diff
@@ -46,42 +35,8 @@ const i18n = new Tannin( {} );
  * @param {LocaleData} [data]   Locale data configuration.
  * @param {string}     [domain] Domain for which configuration applies.
  */
-export function setLocaleData( data, domain = 'default' ) {
-	i18n.data[ domain ] = {
-		...DEFAULT_LOCALE_DATA,
-		...i18n.data[ domain ],
-		...data,
-	};
-
-	// Populate default domain configuration (supported locale date which omits
-	// a plural forms expression).
-	i18n.data[ domain ][ '' ] = {
-		...DEFAULT_LOCALE_DATA[ '' ],
-		...i18n.data[ domain ][ '' ],
-	};
-}
-
-/**
- * Wrapper for Tannin's `dcnpgettext`. Populates default locale data if not
- * otherwise previously assigned.
- *
- * @param {string|undefined} domain   Domain to retrieve the translated text.
- * @param {string|undefined} context  Context information for the translators.
- * @param {string}           single   Text to translate if non-plural. Used as
- *                                    fallback return value on a caught error.
- * @param {string}           [plural] The text to be used if the number is
- *                                    plural.
- * @param {number}           [number] The number to compare against to use
- *                                    either the singular or plural form.
- *
- * @return {string} The translated string.
- */
-function dcnpgettext( domain = 'default', context, single, plural, number ) {
-	if ( ! i18n.data[ domain ] ) {
-		setLocaleData( undefined, domain );
-	}
-
-	return i18n.dcnpgettext( domain, context, single, plural, number );
+export function setLocaleData( data, domain ) {
+	i18n.setLocaleData( data, domain );
```

Comment:
This can also be:
```js
export const setLocaleData = i18n.setLocaleData.bind( i18n );
```

Regarding the JSDoc comments: we don't need to duplicate them at two place. If duplicated, they will almost certainly get out of sync sooner or later.

Reply:
The comments also add type hints so I wouldn't remove them completely. Which copy of the comments would you like me to get rid of? The ones on the class?

Reply:
I'm not sure -- but the comments on the class definitions seem to be the canonical ones. The exported auto-bound functions are just helpers.

What kind of type hints do the comments provide? Is there some editor or IDE that can parse the JSDoc comments and provide some autocomplete hints? Like the TypeScript types do?

Reply:
I'm quite confident that TypeScript and most decent tooling will figure out the corret type of `export const setLocaleData = i18n.setLocaleData.bind( i18n );`

However the documentation tooling didn't discover them in #20332. A may be to help it with something like this, which references the canonical type:

```js
/**
 * @typedef {import('./i18n')} I18n
 */
/**
 * @type {I18n['__']}
 */
export const __ = i18n.__.bind( i18n );
/**
 * @type {I18n['_n']}
 */
export const _n = i18n._n.bind( i18n );
//…
```

Reply:
That fixes the docs but breaks the typings 😬

Reply:
If the `docgen` and TypeScript tools are not happy with the bound reexports without jsdoc comment, then I think it's OK to be explicit and re-introduce the duplicate comments.

Would this work better (after adding TS types and the correct JSDoc)?
```js
/**
 * Retrieve the translation of text.
 * @param ...
 * @returns ...
 */
export const __ = ( text, domain ) => i18n.__( text, domain );
```

Reply:
👍That's basically what I had in the first place 😉